net: openthread: upmerge

edmont opened the pull request:
Update manifest with latest OpenThread upmerge and enable some new configuration options.

rlubos commented on subsys/net/lib/openthread/platform/openthread-core-zephyr-config.h:
```diff
@@ -214,3 +214,49 @@
+ * Set to 1 to enable support for IEEE802.15.4 radio link..
+ *
+ */
+#ifdef CONFIG_OPENTHREAD_RADIO_LINK_IEEE_802_15_4_ENABLE
```
Why aren't the new symbols defined in Kconfig file(s)?

edmont replied:
Yeah, I missed that file. Fixed.

rlubos commented on subsys/net/lib/openthread/platform/openthread-core-zephyr-config.h:
```diff
@@ -214,3 +214,49 @@
+/**
+ * @def OPENTHREAD_CONFIG_RADIO_LINK_IEEE_802_15_4_ENABLE
+ *
+ * Set to 1 to enable support for IEEE802.15.4 radio link..
```
```suggestion
 * Set to 1 to enable support for IEEE802.15.4 radio link.
```

rlubos commented on subsys/net/l2/openthread/Kconfig.thread:
```diff
@@ -66,3 +66,19 @@ config OPENTHREAD_MAX_IP_ADDR_PER_CHILD
+
+config OPENTHREAD_RADIO_LINK_IEEE_802_15_4_ENABLE
+	bool "Enable support for IEEE802.15.4 radio link"
+	default 1
```
`default y`

rlubos commented on subsys/net/l2/openthread/Kconfig.thread:
```diff
@@ -66,3 +66,19 @@ config OPENTHREAD_MAX_IP_ADDR_PER_CHILD
+
+config OPENTHREAD_RADIO_LINK_TREL_ENABLE
+	bool "Thread Radio Encapsulation Link (TREL)"
+	default 0
```
This can be dropped (`n` is default)

rlubos commented:
I get the following build error:
```
In file included from /home/robert/repos/zephyrproject/modules/lib/openthread/src/core/openthread-core-config.h:91,
                 from /home/robert/repos/zephyrproject/modules/lib/openthread/src/cli/cli.hpp:37,
                 from /home/robert/repos/zephyrproject/modules/lib/openthread/src/cli/cli.cpp:34:
/home/robert/repos/zephyrproject/modules/lib/openthread/src/core/config/openthread-core-config-check.h:547:2: error: #error "OPENTHREAD_CONFIG_PLAT_LOG_FUNCTION was replaced by OPENTHREAD_CONFIG_PLAT_LOG_MACRO_NAME " "(and OPENTHREAD_CONFIG_LOG_DEFINE_AS_MACRO_ONLY)"
  547 | #error "OPENTHREAD_CONFIG_PLAT_LOG_FUNCTION was replaced by OPENTHREAD_CONFIG_PLAT_LOG_MACRO_NAME " \
      |  ^~~~~
```

rlubos commented:
The original issue reported is fixed, thanks, but now I get the following:
```
/home/robert/zephyr-sdk/zephyr-sdk-0.11.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/../../../../arm-zephyr-eabi/bin/ld: zephyr/subsys/net/lib/openthread/platform/libopenthread_platform.a(shell.c.obj): in function `ot_cmd':
/home/robert/repos/zephyrproject/zephyr/subsys/net/lib/openthread/platform/shell.c:74: undefined reference to `otCliConsoleInputLine'
/home/robert/zephyr-sdk/zephyr-sdk-0.11.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/../../../../arm-zephyr-eabi/bin/ld: zephyr/subsys/net/lib/openthread/platform/libopenthread_platform.a(shell.c.obj): in function `platformShellInit':
/home/robert/repos/zephyrproject/zephyr/subsys/net/lib/openthread/platform/shell.c:84: undefined reference to `otCliConsoleInit'
collect2: error: ld returned 1 exit status
```
I recall I tried a newer OpenThread some time ago and there was some issue with a new cmake option introduced to OpenThread, I think this might be it: https://github.com/rlubos/openthread/commit/e2f7f18cc1c00465ec97891fb4fff422f11b65b6

edmont commented:
Thanks for catching it @rlubos. I should be fixed now.

rlubos reviewed (approved):
Thank you, it works fine now 👍

rlubos commented on west.yml:
```diff
@@ -111,7 +111,7 @@ manifest:
       path: modules/lib/loramac-node
     - name: openthread
-      revision: f9fbfad0119d7b0d473eaac50c2f0a0a483c2823
+      revision: pull/32/head
```
Please update the SHA: `c3362bc8e24f7e1ec21242c03611595e2d8ec2f0`

edmont committed:
net: openthread: update manifest with latest upmerge

Update manifest to use latest OpenThread.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

edmont committed:
net: openthread: enable new CSL and TREL config options

Enable new OpenThread configuration options:
- OPENTHREAD_CONFIG_RADIO_LINK_IEEE_802_15_4_ENABLE
- OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE
- OPENTHREAD_CONFIG_CSL_SAMPLE_WINDOW
- OPENTHREAD_CONFIG_CSL_RECEIVE_TIME_AHEAD

Update OPENTHREAD_CONFIG_PLAT_LOG_MACRO_NAME

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>
